06-3-ex-sol-ADHD.Rmd

```diff
@@ -0,0 +1,177 @@
+rowdata_modified <- DataFrame(rowdata_modified)
+
+# And then assigned back to the SE object
+rowData(se) <- rowdata_modified
+
+
+# We use this to check what type of data it is
+# read.table(sample_meta_file_path)
+
+# It seems like a comma separated file and it does not include headers
+# Let us read it and then convert from data.frame to DataFrame
+# (required for our purposes)
+sample_meta <- DataFrame(read.table(sample_meta_file_path, sep = ",", header = FALSE))
+
+# Add sample names to rownames
+rownames(sample_meta) <- sample_meta[,1]
+
+# Delete column that included sample names
+sample_meta[,1] <- NULL
+
+# We can add headers
+colnames(sample_meta) <- c("patient_status", "cohort", "patient_status_vs_cohort", "sample_name")
+
+# Then it can be added to colData
+colData(se) <- sample_meta
+
+# Convert to tse format
+tse <- as(se, "TreeSummarizedExperiment")
+
+# Reads the tree file
+tree <- ape::read.tree(tree_file_path)
+
+# Add tree to rowTree
+rowTree(tse) <- tree
+```
+
+ * **Abundance table** Retrieve the taxonomic abundance table from the
+   example data set (TSE object).
+   
+```{r}
+# We show only part of it
+assays(tse)$counts[1:6,1:6]
+```
+ * How many different samples and genus-level groups this phyloseq
+   object has?
+
+```{r}
+dim(rowData(tse))
+```
+
+ * What is the maximum abundance of Akkermansia in this data set?
+
+```{r}
+# Agglomerating to Genus level
+tse_genus <- agglomerateByRank(tse,rank="Genus")
+# Retrieving the count for Akkermansia
+Akkermansia_abund <- assays(tse_genus)$count["Genus:Akkermansia",]
+max(Akkermansia_abund)
+```
+ * Draw a histogram of library sizes (total number of reads per
+   sample).
+
+```{r}
+library(scater)
+tse <- addPerCellQC(tse) # adding a new column "sum" to colData, of total counts/sample.
+hist(colData(tse)$sum, xlab = "Total number of reads per sample", main = "Histogram of library sizes")
+
+```
+ * **Taxonomy table** Retrieve the taxonomy table and print out the
+   first few lines of it with the R command head(). Investigate how
+   many different phylum-level groups this phyloseq object has?
+
+```{r}
+head(rowData(tse))
+taxonomyRanks(tse) # The taxonomic ranks available
+unique(rowData(tse)["Phylum"]) # phylum-level groups
+length(unique(rowData(tse)["Phylum"])[,1]) # number of phylum-level groups
+```
+
+ * **Sample metadata** Retrieve sample metadata. How many patient
+     groups this data set has? Draw a histogram of sample
+     diversities.
+     
+```{r}
+colData(tse) # samples metadata
+unique(colData(tse)$patient_status) # patient groups
+# sample diversity?!?!?!? [MISSING]
+```
+
+ * **Subsetting** Pick a subset of the data object including only
+     ADHD individuals from Cohort 1. How many there are?
+
+```{r}
+sub_cohort_1 <- subsetSamples(tse, colData(tse)$cohort=="Cohort_1")
+sub_cohort_1_ADHD <- subsetSamples(sub_cohort_1, colData(sub_cohort_1)$patient_status=="ADHD")
+colData(sub_cohort_1_ADHD)
```

Comment:
Otherwise looks very good, but I suggest to use [] for subsetting
This should work
```

sub_cohort_1 <- tse[, colData(tse)$cohort=="Cohort_1"]
sub_cohort_1_ADHD <- sub_cohort_1[, colData(sub_cohort_1)$patient_status=="ADHD"]
colData(sub_cohort_1_ADHD)
```

Reply:
ok thank you, i'll add that, anything about sample diversity hist?

Reply:
I think it's good

Reply:
Oh, actually sample diversities might refer here, e.g.,  to patient_status_vs-cohort

Reply:
No, I was meaning the Shannon diversities. Just as a demo. Also other examples can be added if it facilitates understanding of the data.

Reply:
Or I mean bar chart

Reply:
Ok, good

Reply:
done and commited